core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java

```diff
@@ -652,7 +651,7 @@ public void addListener(Object listener, KeyFilter<? super K> filter, ClassLoade
       builder
             .setIncludeCurrentState(l.includeCurrentState())
             .setClustered(l.clustered())
-            .setOnlyPrimary(l.clustered() ? (cacheMode.isDistributed() ? true : false) : l.primaryOnly())
+            .setOnlyPrimary(l.clustered() ? cacheMode.isDistributed() : l.primaryOnly())
```

Comment:
Was this change an oversight or was there some issue? This was specifically for REPL caches so it only notifies on the write to the node where the listener is notified.